consensus/spos/worker.go

```diff
@@ -711,6 +720,89 @@ func (wrk *Worker) ResetConsensusMessages() {
 	wrk.consensusMessageValidator.resetConsensusMessages()
 }
 
+func (wrk *Worker) checkValidityAndProcessEquivalentMessages(cnsMsg *consensus.Message, p2pMessage p2p.MessageP2P) error {
+	wrk.mutEquivalentMessages.Lock()
```

Comment:
can we separate the validation of the final info which becomes equivalent message validation into a separate function?

also would be good if we only add the mutex on the equivalent message validation (after we check the enable flag so we don't affect the previous behavior)

for the equivalent message validation we need to do also aggregated signature verification before we start rejecting other equivalent messages.

Reply:
pushed as discussed:
- now the mutex should only be locked for final info messages, when the flag is active
- added signature verification